### PR TITLE
Various documentation enhancements

### PR DESCRIPTION
**Pull request by thedewi**
See individual commit messages for context.

### CODE REVIEW

**Comment by thedewi** [comment]
@robpearson Could you help me clarify the "or otherwise" on README line 37? Maybe the sentence should just end at "... publish the extension."

https://github.com/OctopusDeploy/OctoTFS/pull/149/files#diff-04c6e90faac2675aa89e2176d2eec7d8R37

**Comment by robpearson** [comment]
> @robpearson Could you help me clarify the "or otherwise" on README line 37? Maybe the sentence should just end at "... publish the extension."
> 
> https://github.com/OctopusDeploy/OctoTFS/pull/149/files#diff-04c6e90faac2675aa89e2176d2eec7d8R37

@thedewi I think that's fine. The rest of the readme describe the steps/procedures so I would leave it as is.

File: source/tasks/OctoInstaller/task.json [hunk 1]
```diff
@@ -36,7 +36,7 @@
             "label": "Octopus Tool Version",
             "defaultValue": "embedded",
             "required": true,
-            "helpMarkDown": "Specify the exact version of the octo tools to use.<br/>"
+            "helpMarkDown": "Specify `latest` or the version number to download. If you specify `embedded`, or the download fails, a built-in copy of the tools will be used."
```
Comment (robpearson) [inline source/tasks/OctoInstaller/task.json]: That's a great change. Nice work @thedewi!

File: source/vsts.md [hunk 2]
```diff
@@ -9,9 +9,13 @@ This extension provides Build and Release tasks to integrate with [Octopus Deplo
 
 You will need a minimum build agent version of `2.115.0` with .NET Core SDK `2.0` or later. When targeting build agents without the SDK, you can use the **.NET Core SDK Installer** task to install it. Generally the Hosted Linux, Mac and Hosted VS2017 agent queues already provide it, however please refer to Microsoft documentation regarding what capabilities are provided by which hosted agent pools.
 
-The Octopus tasks will automatically download, cache, and use the latest version of the Octopus tools, unless they are supplied using the **Octopus tools installer** task, or found using the `PATH` environment variable.
+## Add Octo.exe Command Line tools
 
-If the **Octopus tools installer** task version is set to `embedded`, it will supply an embedded copy of the tools instead of downloading.
+The Octopus tasks require the Octo.exe Command Line tools, which can be supplied any of the following ways:
```
Comment (robpearson) [inline source/vsts.md]: I'm being picky but I'd make it `tool`. Singular vs plural. I think of `octo.exe` as our command line too or command line interface (CLI).

File: source/vsts.md [hunk 3]
```diff
@@ -9,9 +9,13 @@ This extension provides Build and Release tasks to integrate with [Octopus Deplo
...
+
+* Insert the **Octopus tools installer** task into the build pipeline. If you leave the version as `embedded`, a built-in copy of the tools will be supplied to the other tasks. If you set the version to `latest` or a specific number, that version will be downloaded and supplied to the other tasks. If the download fails, the built-in copy of the tools will be used.
```
Comment (robpearson) [inline source/vsts.md]: I'd probably say `Add` vs `Insert` as I think that's a bit more accurate. Food for thought.

Comment (robpearson) [inline source/vsts.md]: Maybe add an example of a specific number ... `or a specific number like 6.7.0, that version ...`

File: source/vsts.md [hunk 4]
```diff
@@ -9,9 +9,13 @@ This extension provides Build and Release tasks to integrate with [Octopus Deplo
...
+* Update the system `PATH` environment variable to include a folder containing `Octo.exe` or `Octo.cmd`, on all systems running VSTS agents. You will need to restart the `VSTS Agent` service (or the whole system) for the change to take effect.
+* If the Octo.exe Command Line tools are not supplied, the Octopus tasks will attempt to download the latest version themselves. **Note**: For technical reasons they cannot access the built-in copy of the tools unless the **Octopus tools installer** task already ran.
```
Comment (robpearson) [inline source/vsts.md]: I'd remove the note about for technical reasons. Simpler for our customers.

File: source/vsts.md [hunk 5]
```diff
@@ -86,9 +92,9 @@ Options include:
 
  * **Octopus Deploy Server**: The Octopus Server (click **New** to [add a service connection](#Add-a-service-connection-to-Octopus-Deploy)).
  * **Space**: The Octopus space to push package metadata to.
- * **Package ID**: The ID of the package, pushed previously, to push metadata onto. e.g. MyCompany.App
- * **Package Version**: The version of the package, pushed previously, to push metadata onto.
- * **Work Items Source**: The service hosting any work items associated with each version of the package. Octopus will add information about the work items to the package metadata, which can be used in release notes.
+ * **Package ID**: The ID of the package, pushed separately, to push metadata onto. e.g. MyCompany.App
+ * **Package Version**: The version of the package, pushed separately, to push metadata onto.
+ * **Work Items Source**: The service hosting any work items or issues associated with each version of the package. Octopus will add information about the work items or issues to the package metadata, which can be used in release notes.
```
Comment (robpearson) [inline source/vsts.md]: Should we link to the docs here?

File: source/vsts.md (hunk 2 above)
Comment (thedewi) [reply source/vsts.md]: Ok, done.

File: source/vsts.md (hunk 3 above)
Comment (thedewi) [reply source/vsts.md]: Changed to 
```markdown
Add the **Octopus tools installer** task to the build pipeline, ahead of other Octopus tasks.
```

Comment (thedewi) [reply source/vsts.md]: Agreed.

File: source/vsts.md (hunk 4 above)
Comment (thedewi) [reply source/vsts.md]: Ok.

File: source/vsts.md (hunk 5 above)
Comment (thedewi) [reply source/vsts.md]: Good idea.

**Pull request merged**